daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -279,6 +279,18 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
 
   def roots: ImmArray[Nid]
 
+  // TODO: https://github.com/digital-asset/daml/issues/8020
+  //  for now we assume that rollback node cannot be a root of a transaction.
```

Comment:
I thought the plan was to keep assuming that because the only case where this is not the case is projections and we handle those separately?

Reply:
Yes. I just want to keep track of this and eventually reconsider it before releasing #8020.